Fixing folder creation on downloads

martinjuhasz opened the pull request:
Download commands should not create a `screenshots` or `metadata` folder if a custom path is used, something that i missed yesterday.

If you use the options `screenshots_path` or `metadata_path` for uploading, you specify the path directly pointing to the related directory, but if you set this for downloading it creates another `screenshot`/`metadata` directory inside.

The classes/methods for uploading take a path param and use this path without appending something. The download classes now do the same and the appending of the correct subfolder happens before calling it.

I think this is now the behavior that was intended?

KrauseFx commented:
Why does this require so many changes when it's basically just fixing folder not there yet?

martinjuhasz commented:
Its more about appending the `screenshots` and `metadata` folder if no custom path was set, than creating missing folders. `generate_metadata_files` and `download_screenshots` always appended these no matter if a custom download/metadata path was set or not. They no longer do this and if a folder should created moved to the `command_generator`, like it was done for uploading.

martinjuhasz committed:
Fixing folder creation on downloads

Download commands should not create a `screenshots` or `metadata`
folder if a custom path is used.

martinjuhasz commented:
don't know why squashing is always such a hassle for me. anyway, fixed a typo and pushed again.

asfalcone commented:
Thanks @martinjuhasz! :open_file_folder: :sparkles:
